src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```diff
@@ -2004,10 +2003,10 @@ private static String getExtractedRuleName(String ruleName) {
 
         for (String part : parts) {
             if (part.matches("[A-Za-z]+")) {
-                // Capitalize the first letter and make the rest words
+                // Capitalize the first letter and make the rest words lowercase
                 extractedRuleName
                         .append(part.substring(0, 1).toUpperCase(Locale.ENGLISH))
-                        .append(part.substring(1));
+                        .append(part.substring(1).toLowerCase(Locale.ENGLISH));
```

Comment:
had to modify this section based on @romani's suggestions here: https://github.com/checkstyle/checkstyle/pull/15094#discussion_r1649704947